test/test.cpp

```diff
@@ -36,7 +36,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 unit_test_t _g_unit_tests[1024];
 int _g_num_unit_tests = 0;
-int _g_test_failures = 0;
+int _g_test_failures = 0; // flushed at start of every unit
+int _g_test_total_failures = 0;
```

Comment:
you right, it shadowed real failures count by last successfull result.
test framework already had some total_failures counter so i just moved it to global scope.